src/OSPSuite.UI/Views/Charts/DataBrowserView.cs

```diff
@@ -40,6 +40,20 @@ public DataBrowserView()
          InitializeWith(gridView);
          _gridViewBinder = new GridViewBinder<DataColumnDTO>(gridView);
          _usedRepository = new UxRepositoryItemCheckEdit(gridView);
+
+         
+         gridView.CustomDrawGroupRow += (s, e) =>
+         {
+            var info = e.Info as GridGroupRowInfo;
+
+            foreach (var replacementPair in GroupingNamesReplacementDictionary)
```

Comment:
This is very, very inefficient. I know you only have one entry in the group at the moment, but it's a key value pair where key is the way to look for an entry. Right now, you are treating it as a simple array, looping over everything, not breaking when found. 
I am not sure that we need this method but if we do, it needs to be rewritten.e.g.
something along the line of
```
if(GroupingNamesReplacementDictionary.HasKey((info.GroupValueText))
   info.GroupText = replacementPair.Value;
```

no foreach loop.

Reply:
I have changed this. I have also originally thought of implementing this as  List<string, string> or with Tuples (or a new class for that matter.). We could briefly discuss this today in the meeting.